src/components/Avatar/types/main.ts

```diff
@@ -0,0 +1,25 @@
+import type React from 'react';
+
+import type {DistributiveOmit} from '../../../types/utils';
+import type {DOMProps, QAProps} from '../../types';
+import type {AvatarIconProps} from '../AvatarIcon';
+import type {AvatarImageProps} from '../AvatarImage';
+import type {AvatarTextProps} from '../AvatarText';
+
+import type {AvatarCommonProps, AvatarSize} from './common';
+
+export type AvatarTheme = 'normal' | 'brand';
+export type AvatarView = 'filled' | 'outlined';
+
+interface AvatarBaseProps extends DOMProps, QAProps {
+    size?: AvatarSize;
+    theme?: AvatarTheme;
+    view?: AvatarView;
+    backgroundColor?: string;
+    borderColor?: string;
+    title?: string;
+    ariaAttributes?: React.AriaAttributes;
+}
+
+export type AvatarProps = AvatarBaseProps &
```

Comment:
We don't need `AvatarCommonProps` if each subtype will extend from `AvatarBaseProps`, then we can write final type like:
```ts
type AvatarProps = AvatarImageProps | AvatarIconProps | AvatarTextProps
```

Reply:
But subcomponents (AvatarIcon, AvatarText and AvatarImage) don't need props like theme, view and so on. `AvatarCommonProps` only for subcomponents